### DISCUSSION TITLE
Group Mapping not working with OAuth2/OIDC with AzureAD/EntraID (groups claim)

### DISCUSSION THREAD

**Discussion by maholick** [requarks/wiki#6894] [Error / Bug Report]
Hi there,

I'm trying to get the "Map Groups" function working within the authentication methods "OAuth2" and/or "Generic OpenID Connect / OAuth2" with Entra ID (former Azure AD). 

WikiJS System Info:
<img width="1198" alt="image" src="https://github.com/requarks/wiki/assets/57866444/b6cf21f0-7ed5-4796-8690-e2543c8a280a">

I could not find an official documentation for WikiJS about how to configure it, but I think I tried everything I could to get this work.
The authentication is working like a charm, but the groups which are configured with the application in Azure is not mapping the user to the groups in WikiJS. 

I tried it with the group names and also with the objectid as group name in WikiJS without success.
I also tried different token / claim configurations in Azure without success.

I also tried to debug the JWT token and it only gives me the number of groups, but no group names. The payloads comes from WikiJS I think and not sure if this should inlcude the group names (probably not).

Payload:

>  "email": "shawn.xxx@xyz.com",
>   "name": "Shawn XXX",
>   "av": "",
>   "tz": "America/New_York",
>   "lc": "en",
>   "df": "",
>   "ap": "",
>   "permissions": [
>     "read:pages",
>     "read:assets",
>     "read:comments"
>   ],
>   "groups": [
>     3
>   ],
>   "iat": 1702810698,
>   "exp": 1702812498,
>   "aud": "urn:wiki.js",
>   "iss": "urn:wiki.js"
> }

### **Here my configuration:**

**WikiJS**
 
<img width="910" alt="image" src="https://github.com/requarks/wiki/assets/57866444/2b5bac6f-56e0-4c16-a6b0-548ac765758c">

**EntryID / AzureAD**

<img width="1273" alt="image" src="https://github.com/requarks/wiki/assets/57866444/52b0e3d2-f069-4327-a077-e14c775368dc">

<img width="939" alt="image" src="https://github.com/requarks/wiki/assets/57866444/84a013b8-3833-49a4-8560-2ca8e9a5f785">

<img width="684" alt="image" src="https://github.com/requarks/wiki/assets/57866444/0b75ed0b-3808-4161-8998-1182a6a64d20">

<img width="619" alt="image" src="https://github.com/requarks/wiki/assets/57866444/82f9f210-588b-443a-b9c5-c97fadfc51f7">

The groups which should be mapped are 1. added to the application and 2. created within WikiJS (as name and with objectid).

Does someone has an idea or can confirm that his might me an error or bug?
Also further documentation or manuals would be helpful.

**Comment by dannykorpan** [comment]
Anybody?

**Comment by lacsapztem** [comment]
Hello 

It is probably way too late for OP, but for anybody having this problem, I have solved it by setting sAMAccountName in group claim instead of "Group Id"
And I had to manually create a group with the same name in Wiki.js
<img width="500" alt="image" src="https://github.com/user-attachments/assets/e75658a5-c364-4b88-8ff8-338510748163" />

**Reply by Hallo5000** [reply]
Somewhat same for me (took a while till I noticed that groups will only be applied if they already exist in WikiJs). 
Also leave "Assign to group" empty if you want the users to only have their groups from the authentication provider.
